File: src/main/java/grojdg/harderstats/events/PlayerManagerEvents.java
package grojdg.harderstats.events;

import grojdg.harderstats.HarderStats;
import grojdg.harderstats.InfoReceptionService;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.world.World;

public class PlayerManagerEvents {

    // start back up the nether timer if the player joins and spawns into the nether
    public static void afterPlayerConnect(ServerPlayerEntity player) {
        HarderStats.LOGGER.info("afterPlayerConnect is in world: " + player.getWorld().getRegistryKey());
        if (player.getWorld().getRegistryKey() == World.NETHER) {
            InfoReceptionService.InfoReceptionServiceFactory.get().setIsInNether(player.getUuid(), true);
        }
    }
}

File: src/main/kotlin/grojdg/harderstats/events/PlayerManagerEvents.kt
package grojdg.harderstats.events

import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.server.network.ServerPlayerEntity

object PlayerManagerEvents {
    fun interface AfterPlayerConnect {
        fun interact(player: ServerPlayerEntity)
    }

    val AFTER_PLAYER_CONNECT = EventFactory.createArrayBacked(AfterPlayerConnect::class.java
    ) { listeners ->
        AfterPlayerConnect { player ->
            for (listener in listeners) {
                listener.interact(player)
            }
        }
    }
}